packages/orbit-tailwind-preset/src/presets/components/index.ts

```diff
@@ -66,191 +70,194 @@ const getBackgroundColors = () => {
 
 const cfg = (options?: Options): Config => {
   const { disablePreflight = true } = options || {};
+  // make palette colors (foundation colors) defined as css vars and make components tokens inherit it
+  const theme = getTokens(cssVarsFoundation);
+  const componentTokens = getComponentLevelToken(theme);
 
   return {
     content: ["auto"],
-    presets: [orbitFoundationPreset],
+    presets: [orbitFoundationPreset(theme)],
     corePlugins: {
       preflight: disablePreflight ? false : undefined,
     },
     theme: {
+      variables: {
+        // paletteBlueNormal -> --palette-blue-normal
+        DEFAULT: Object.entries(theme).reduce((acc, [key, value]) => {
```

Comment:
This feels like we're going round in circles. We have an object defined that matches keys to CSS vars. We create a new object with camelCase formatted tokens as keys and the same CSS vars as values. And then we iterate over the same object to convert the keys to kebab-case and keep the same values.

Can't a step be skipped here? 

I understand that maybe this is caused by the need of having some tailwind classes to use values directly and others to have CSS Vars, but maybe we can improve this. Colors to one side and everything else to the other might be a solution here. Since the only customizable thing in our themes are colors, we can use that in our favor

Reply:
*Agreed in DMs, that it will be postponed on further larger refactoring*